### PR TITLE
[BugFix] Bind storage volume to db when it has been created successfully

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue
```
MySQL [(none)]> create database test properties("storage_volume"="storage_volume4");
ERROR 1064 (HY000): Unexpected exception: Unknown storage volume "storage_volume4"
MySQL [(none)]> create database test properties("storage_volume"="storage_volume4");
ERROR 1064 (HY000): Unexpected exception: Can't create database 'test'; database exists
MySQL [(none)]>show databases;
+--------------------+
| Database           |
+--------------------+
| _statistics_       |
| information_schema |
| starrocks          |
| test              |
+--------------------+
5 rows in set (0.01 sec)
```
database will be created if the storage volume not exists.
database should be created after checking the existence of storage volume and the edit log is written successfully.
## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|7|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -4605,7 +4608,6 @@ private String bindStorageVolumeToDb(long dbId, String volume) throws DdlExcepti
             throw new DdlException("Unknown storage volume \"" + volume + "\"");
         }
         String storageVolumeId = sv.getId();
-        svm.bindDbToStorageVolume(storageVolumeId, dbId);
         return storageVolumeId;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4611]: `return sv.getId();`

**Review by wyb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|8|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -381,18 +381,21 @@ public void createDb(String dbName, Map<String, String> properties) throws DdlEx
...
                     }
-                    storageVolumeId = bindStorageVolumeToDb(id, volume);
+                    storageVolumeId = getStorageVolumeId(volume);
                 }
+                id = getNextId();
+                Database db = new Database(id, dbName);
+                unprotectCreateDb(db);
                 GlobalStateMgr.getCurrentState().getEditLog().logCreateDb(db, storageVolumeId);
+                if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+                    GlobalStateMgr.getCurrentState().getStorageVolumeMgr().bindDbToStorageVolume(storageVolumeId, id);
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:397]: what if the storageVolume exists in line 390, but removed in other threads when running to line 397?

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 15 / 19 (78.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|9|13|69.23%|[397, 401, 402, 403]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|6|6|100.00%|[]|

**Review by HangyuanLiu — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 12 (91.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|6|83.33%|[395]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**